src/components/Form.js

```diff
@@ -54,18 +54,43 @@ export default class Form extends Component {
     this.setState(nextState);
   }
 
-  getStateFromProps(props, inputFormData) {
+  getStateFromProps(nextProps, inputFormData) {
     const state = this.state || {};
-    const schema = "schema" in props ? props.schema : this.props.schema;
-    const uiSchema = "uiSchema" in props ? props.uiSchema : this.props.uiSchema;
+
+    if (
+      nextProps &&
+      inputFormData === state.formData &&
+      nextProps.schema === this.props.schema &&
+      nextProps.uiSchema === this.props.uiSchema &&
+      nextProps.liveValidate === this.props.liveValidate &&
+      nextProps.customFormats === this.props.customFormats &&
+      nextProps.additionalMetaSchemas === this.props.additionalMetaSchemas &&
+      nextProps.noValidate === this.props.noValidate &&
+      nextProps.idPrefix === this.props.idPrefix
+    ) {
+      // No data has changed, maintain state.
+      return state;
+    }
+
+    const schema = nextProps ? nextProps.schema : this.props.schema;
+    const uiSchema = nextProps ? nextProps.uiSchema : this.props.uiSchema;
+    const liveValidate = nextProps
```

Comment:
Is there a particular reason why you changed the style here (from something like `props.liveValidate || this.props.liveValidate` to `nextProps ? nextProps.liveValidate : this.props.liveValidate`)?

Reply:
If liveValidate changed from true to false, props.liveValidate would pass over to this.props.liveValidate and liveValidate would remain on despite being turned off by the caller.